### ISSUE TITLE
feat: Run command after generating brick

### ISSUE THREAD

**Issue by KlausJokisuo** [felangel/mason#114]
**Description**

Run command after generating brick. The command would also accept parameters.

**Description**

I would use mason to generate folder structures and generic boilerplate for Flutter applications.
I myself and quite many other people rely on some sort of code generation, so it would be great if we could run code generation after the brick content is generated.

**Comment by felangel** [comment]
Hi @KlausJokisuo 👋
Thanks for opening an issue!

Can you provide a bit more context? I’m not sure I fully understand the issue. If you could provide an example as well that’d be super helpful, thanks!

**Comment by KlausJokisuo** [comment]
@felangel My bad, forgot to write the description. I've updated the OP :)

**Comment by marcossevilla** [comment]
hi @KlausJokisuo! We're evaluating what can be the best approach for your request and the following proposal came up.

Having an extra folder created with template files in which you can add your custom pre/post-generation scripts. The structure would look like this:
```
├── __brick__
│   ├── HELLO.md
│   ├── {{~\ footer.md\ }}
│   └── {{~\ header.md\ }}
├── __hooks__
│   ├── post_gen.sh
│   └── pre_gen.sh
└── brick.yaml
```
We can also support an structure where a Dart script is executed but I think it's better to have it generalized with a Bash script.

A Dart script can be supported, similar to what [`mason bundle`](https://github.com/felangel/mason#bundle-usage) does by specifying if it's Dart or universal.

Let us know if that aligns with your needs and hopefully, more people from the community can share their opinions here. 👍

**Comment by felangel** [comment]
Closed by #149 🎉
